### ISSUE TITLE
Webpack 5 Compatibility

### ISSUE THREAD

**Issue by sectsect** [cascornelissen/svg-spritemap-webpack-plugin#108]
Thanks for helpful package.

Currently this package does not work with webpack 5 beta.
Do you have a plan to add compatibility before webpack 5 is officially released?

**Comment by cascornelissen** [comment]
I'm planning to look into support Webpack 5 when the first RC, yes. I'll leave this issue open to track progress once a RC is available.

Others are of course free to investigate this before a RC is released.

**Comment by sectsect** [comment]
@cascornelissen 

Hi.
webpack 5 is now in RC.
```js
"webpack": "^5.0.0-rc.3",
```

and the release planned for 2020-10-10.
https://github.com/webpack/webpack/issues/11406

I can't wait for added webpack 5 compatibility to this package.

Thanks.

**Comment by cascornelissen** [comment]
I've noticed and have been playing around with it for a bit but it seems like the `MultiEntryPlugin` which is used by this plugin is no longer available in webpack 5 and I can't find any information about it so until there is some migration documentation I'm holding of from continuing on this for now.

**Comment by sectsect** [comment]
@cascornelissen 

I got a useful tip.
we can use `webpack/lib/EntryPlugin` multiple times instead of `MultiEntryPlugin` on webpack5.

https://twitter.com/wSokra/status/1311701299914776576?s=20

**Comment by cascornelissen** [comment]
PSA: I'm aware Webpack 5 was released earlier today and am still working on getting this plugin to work with both 4 and 5.

**Comment by cascornelissen** [comment]
Alright, this took way too long for the simplicity of the required change that was blocking a release but thankfully the man, the myth, the legend, Tobias Koppers himself, stepped in when I create an issue on the GitHub repository. His comment pushed me in the right direction and we now have a new release on NPM under the `next` tag. Install it using the following command:

```
$ npm install svg-spritemap-webpack-plugin@next
```

Please test this release out using both Webpack 4 and 5 if possible and report any problems. If nothing pops up during the coming days I'll release it with the `latest` dist-tag.

**Comment by sectsect** [comment]
@cascornelissen 

**v3.6.0** create `spritemap.js` file.
Do you have option to delete this file?

Related? #17 

```js
(self["webpackChunkwebpack_postcss"] = self["webpackChunkwebpack_postcss"] || []).push([["spritemap"],{

/***/ "?4e0c":
/*!******************************!*\
  !*** spritemap-dummy-module ***!
  \******************************/
/*! unknown exports (runtime-defined) */
/*! runtime requirements:  */
/***/ (function() {



/***/ })

}]);
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoic3ByaXRlbWFwLmpzIiwic291cmNlcyI6W10sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7Ozs7QTs7QSIsInNvdXJjZVJvb3QiOiIifQ==
```

**Comment by cascornelissen** [comment]
@sectsect, thanks for reporting the issue, it should be delete automatically (unless `chunk.keep` is set to `true`), there's even a test for it which seems to be green for both Webpack 4 and 5 so I'll investigate manually.

**Comment by markgaze** [comment]
The `#webpack5` version fixed my issues but since switching to v3.6.0, I get this:

```
node_modules/svg-spritemap-webpack-plugin/svg4everybody-helper.js 30 bytes [built] [code generated]
node_modules/svg4everybody/dist/svg4everybody.js 6.03 KiB [built] [code generated]
spritemap-dummy-module 1 bytes [code generated]
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
ERROR in requestWithoutMatchResource.charCodeAt is not a function
webpack 5.3.0 compiled with 21 errors in 1522 ms
```

I've tried to change a few things to see if there was anything else causing it (as I've bumped a few other dependencies at the same time) but it seems like the update to v3.6.0 is the cause. Let me know if I can help you out with debugging this and I'll see what I can do.

**Comment by cascornelissen** [comment]
Interesting, could you (and @sectsect as well, and anyone running into other issues with `3.6.x`) create a new issue? I wasn't expecting multiple issues to pop up but that'll hopefully prevent some confusion.

For anyone creating issues: it helps a lot if you provide a minimal configuration that reproduces the issue, especially if I can just copy/paste that configuration into the [simple example](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/tree/master/examples/simple).

**Comment by cascornelissen** [comment]
All bugs that were found so far have been fixed and released in `3.6.1`.

```
$ npm install svg-spritemap-webpack-plugin@next
```

Please test this release out using both Webpack 4 and 5 if possible and report any problems. If nothing pops up during the coming days I'll release it with the `latest` dist-tag this Sunday.

**Comment by cascornelissen** [comment]
With the release of [`3.6.2`](https://github.com/cascornelissen/svg-spritemap-webpack-plugin/releases/tag/v3.6.2) on NPM with the `latest` dist-tag: Webpack 5 is now officially supported 🚀 

If anyone runs into anything, please create a new issue.
